XenAdmin/Dialogs/NetworkingPropertiesPage.cs

```diff
@@ -166,15 +167,18 @@ public void RefreshButtons()
                 haEnabledRubric.Enabled =
                     true;
             }
-            if (network != null)
+
+            var pif = (PIF) Tag;
+            var existingCluster = network != null ? network.Connection.Cache.Clusters.FirstOrDefault() : null;
+
+            if (pif != null && existingCluster != null && existingCluster.network.opaque_ref == network.opaque_ref)
             {
-                var master = Helpers.GetMaster(network.Connection);
+                Host host = network.Connection.Resolve(pif.host);
+                    
                 var clusteringEnabled = network.Connection.Cache.Cluster_hosts.Any(cluster =>
-                    cluster.host.opaque_ref == master.opaque_ref && cluster.enabled);
-                var existingCluster = network.Connection.Cache.Clusters.FirstOrDefault();
-
-                if (clusteringEnabled && existingCluster != null &&
-                    existingCluster.network.opaque_ref == network.opaque_ref)
+                    cluster.host.opaque_ref == host.opaque_ref && cluster.enabled);
+                    
+                if (clusteringEnabled && host.enabled)

```

Comment:
It is normally a good idea to check the host for null in case it hasn't resolved.